pkg/plugins/policies/core/xds/meshroute/clusters.go

```diff
@@ -91,6 +91,11 @@ func GenerateClusters(
 					}
 				} else {
 					if service.BackendRef().ReferencesRealObject() {
+						if service.BackendRef().Kind == common_api.MeshService {
+							if ms := meshCtx.MeshServiceByName[service.BackendRef().Name]; ms != nil {
```

Comment:
I think we might not have `.Name` set right? If do backendRefs with labels

Reply:
that's true. This is broader issue https://github.com/kumahq/kuma/issues/10947